src/ser/mod.rs

```diff
@@ -51,6 +63,8 @@ pub type Result<T> = std::result::Result<T, Error>;
 /// Serialization error.
 #[derive(Clone, Debug, PartialEq)]
 pub enum Error {
+    /// io error from std::io
+    IoError(String),
```

Comment:
this is changing the public API - a breaking change

Reply:
Yep, I did it like this because that's how it's done in `de` but I don't mind doing it some other way

Reply:
Is it a breaking change because people matching on `Error` would have to match additionally on `IoError`? 🤔

Reply:
yes, adding a variant to an exhaustive public enum is always a breaking change